Sources/Rendering/Core/CellPicker/index.js

```diff
@@ -263,31 +275,47 @@ function vtkCellPicker(publicAPI, model) {
     const locator = null;
     if (locator) {
       // TODO when cell locator will be implemented
-    } else if (data.getPolys) {
-      const cellObject = data.getPolys();
-      const points = data.getPoints();
-      const cellData = cellObject.getData();
-      let cellId = 0;
-      const pointsIdList = [-1, -1, -1];
-      const cell = vtkTriangle.newInstance();
-      const cellPoints = vtkPoints.newInstance();
-      // cross all cells
-      for (let i = 0; i < cellData.length; cellId++) {
+    } else if (data.getCells) {
+      if (!data.getCells()) {
+        data.buildLinks();
+      }
+
+      const tempCellMap = createCellMap();
+      const minCellMap = createCellMap();
+
+      const numberOfCells = data.getNumberOfCells();
+
+      for (let cellId = 0; cellId < numberOfCells; cellId++) {
         const pCoords = [0, 0, 0];
-        const nbPointsInCell = cellData[i++];
 
-        cellPoints.setNumberOfPoints(nbPointsInCell);
-        // Extract cell points
-        for (let j = 0; j < nbPointsInCell; j++) {
-          pointsIdList[j] = cellData[i++];
+        minCellType = data.getCellType(cellId);
+        const cell = tempCellMap[minCellType];
+
+        if (cell === undefined) {
```

Comment:
`if (cell == null)` is more flexible

Reply:
Sure but it will never be null, or am I missing something?
If it's null it's because one of the `newInstance`calls returned null so in this case I think it's better to have an error than silently fail.

Reply:
To me it is not this function responsibility to trigger such error.
However, this function needs to support such scenario and not crash.